### PR TITLE
Added path-ignore to workflows for documentation or workflow changes.

### PR DESCRIPTION
**Pull request by jefchien**
**Description:** The workflows don't need to be triggered for changes that do not affect the behavior of the components.

### CODE REVIEW

**Comment by jefchien** [comment]
Might make sense to still trigger the workflows on push.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -4,7 +4,13 @@ on:
     branches: [ main ]
     tags:
       - 'v[0-9]+.[0-9]+.[0-9]+*'
+    paths-ignore:
+      - '.github/**'
```
Comment (codeboten) [inline .github/workflows/build-and-test.yml:8]: how would this work if someone is making a change to build-and-test that needs to be run in order to see if it works?

**Comment by anuraaga** [comment]
Generally this setting doesn't work since required checks won't be run on docs PRs and the PRs become unmergable (similar to https://github.com/aws-observability/aws-otel-collector/pull/709#issuecomment-960377203 but unlike that repo, where it probably needs to be fixed, here the status checks are actually required I believe).

**Comment by jefchien** [comment]
Only the `build-and-test` workflow seems to be required. I'll look into just fast-succeeding the steps for `build-and-test`. Are the other workflows fine to skip?

**Pull request closed without merging**